master/internal/grpcutil/auth.go

```diff
@@ -41,8 +46,30 @@ var (
 	ErrPermissionDenied = status.Error(codes.PermissionDenied, "user does not have permission")
 )
 
-// GetAllocationSession returns the currently running task.
-func GetAllocationSession(ctx context.Context, d *db.PgDB) (*model.AllocationSession, error) {
+func allocationSessionByTokenBun(token string) (*model.AllocationSession, error) {
+	v2 := paseto.NewV2()
+
+	var session model.AllocationSession
+	err := v2.Verify(token, db.GetTokenKeys().PublicKey, &session, nil)
+	if err != nil {
+		log.WithError(err).Debug("failed to verify allocation_session token")
+		return nil, db.ErrNotFound
+	}
+
+	err = db.Bun().NewSelect().Model(&session).Where("id = ?", session.ID).Scan(context.Background())
+	if errors.Cause(err) == db.ErrNotFound {
```

Comment:
bun will return something like ```sql.ErrNoRows``` so this ```db.ErrNotFound``` will never go off though in this case it is super unlikely to happen but something to keep in mind as we move towards bun.

Reply:
good catch